devito/ir/iet/nodes.py

```diff
@@ -742,14 +742,21 @@ class TimedList(List):
     def __init__(self, timer, lname, body):
         self._name = lname
         self._timer = timer
-        header = [c.Statement("struct timeval start_%s, end_%s" % (lname, lname)),
-                  c.Statement("gettimeofday(&start_%s, NULL)" % lname)]
-        footer = [c.Statement("gettimeofday(&end_%s, NULL)" % lname),
-                  c.Statement(("%(gn)s->%(ln)s += " +
-                               "(double)(end_%(ln)s.tv_sec-start_%(ln)s.tv_sec)+" +
-                               "(double)(end_%(ln)s.tv_usec-start_%(ln)s.tv_usec)" +
-                               "/1000000") % {'gn': timer.name, 'ln': lname})]
-        super(TimedList, self).__init__(header, body, footer)
+
+        super().__init__(header=c.Line('START_TIMER(%s)' % lname),
+                         body=body,
+                         footer=c.Line('STOP_TIMER(%s,%s)' % (lname, timer.name)))
+
+    @classmethod
+    def _start_timer_header(cls):
+        return ('START_TIMER(S)', ('struct timeval start_ ## S , end_ ## S ; '
+                                   'gettimeofday(&start_ ## S , NULL);'))
+
+    @classmethod
+    def _stop_timer_header(cls):
```

Comment:
Should this not be _stop_timper_footer?

Reply:
good catch. Since this appears to be the only (tiny) oversight of this PR, I'll fix it in the next imminent PR